src/js/player.js

```diff
@@ -2218,27 +2219,23 @@ class Player extends Component {
    *         The current video source when getting
    */
   src(source) {
-    if (source === undefined) {
-      return this.cache_.src;
-    }
-
-    this.changingSrc_ = true;
-
     let src = source;
 
-    if (Array.isArray(source)) {
-      this.cache_.sources = source;
-      src = source[0];
-    } else if (typeof source === 'string') {
-      src = {
-        src: source
-      };
+    // filter out invalid sources and turn our source into
+    // an array of source objects
+    src = filterSource(src);
 
-      this.cache_.sources = [src];
+    // no valid sources, return the current source
+    if (!src.length) {
+      return this.cache_.src;
     }
 
-    this.cache_.source = src;
+    this.cache_.sources = src;
+    src = src[0];
```

Comment:
Yes, but previously `source` was the source of truth for a list of sources. With the `filterSource` functionality, the output of `filterSource` became the source of truth.